src/main/kotlin/no/item/xp/plugin/util/GenerateInterfaceName.kt

```diff
@@ -0,0 +1,22 @@
+package no.item.xp.plugin.util
+
+import java.io.File
+
+@Suppress("FunctionName")
+fun GenerateInterfaceName(fileName: File): String? {
+  val filePath: String? = pathComponent(fileName.absolutePath)
+  val newFileName: String = filePath + "\\" + fileName.nameWithoutExtension + ".ts"
+  val file = File(newFileName)
+  val isNewFileCreated: Boolean = file.createNewFile()
+  return if (isNewFileCreated) {
+    newFileName
+  } else {
+    println("$newFileName already exists.")
+    null
```

Comment:
Lets not return `null` but return an `Option` instead.

Reply:
Changes made to branch: TypeScriptWriterImplementation. Solved with implementing an Either<Throwable, String>